javascript/features/player_settings/player_settings_manager.js

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Las Venturas Playground. All rights reserved.
+// Use of this source code is governed by the MIT license, a copy of which can
+// be found in the LICENSE file.
+import { PlayerSettingsDatabase } from 'features/player_settings/player_settings_database.js';
+
+import { MockPlayerSettingsDatabase } from 'features/player_settings/test/mock_player_settings_database.js';
+
+export class PlayerSettingsManager {
+    constructor() {
+        // The database instance used to read and write persistent values.
+        this.database_ = server.isTest() ? new MockPlayerSettingsDatabase()
+                                         : new PlayerSettingsDatabase();
+
+        server.playerManager.addObserver(this, true /* replayHistory */);
+    }
+
+    // Loads the settings of the player when a player has logged in
+    onPlayerLogin(player) {
+        if (!player.isRegistered()) {
+            return;
+        }
+
+        // Load the existing persistent values from the database, and apply them to the local state.
+        Promise.resolve(this.database_.loadSettings(player.userId)).then(settings => {
+            for (const [identifier, value] of settings) {
+                try {
+                    this.setValue(identifier, value);
+                } catch (exception) {
+                    console.log(`Warning: Unable to restore the player setting of ${identifier}: ` +
```

Comment:
I'm not sure if this'll produce sensible output, because it depends on the toString() implementation of the exception. Passing |exception| as a separate parameter may be more clear.